frontend/app/utils/CampaignCode.scala

```diff
@@ -8,9 +8,8 @@ case class CampaignCode(get: String)
 object CampaignCode {
   def fromRequest(implicit request: RequestHeader): Option[CampaignCode] =
     for {
-      cookie <- request.cookies.get("s_sess")
-      decodedCookie = UriEncoding.decodePathSegment(cookie.value, "utf-8")
-      regexMatch <- "campaign=(.+?);".r.findFirstMatchIn(decodedCookie)
-      code <- Option(regexMatch.group(1))
-    } yield CampaignCode(code)
+      cookie <- request.cookies.get("mem_campaign_code")
+    } yield {
+    	CampaignCode(cookie.value)
```

Comment:
Is this 100% safe? The cookie may not be present, so will cookie.value throw an exception?